modules/swagger-codegen/src/main/resources/typescript-inversify/api.service.mustache

```diff
@@ -0,0 +1,177 @@
+        }
+        {{/isListContainer}}
+        {{^isListContainer}}
+        if ({{paramName}}) {
+            headers['{{baseName}}'] = String({{paramName}});
+        }
+        {{/isListContainer}}
+{{/headerParams}}
+
+{{#authMethods}}
+        // authentication ({{name}}) required
+{{#isApiKey}}
+{{#isKeyInHeader}}
+        if (this.APIConfiguration.apiKeys["{{keyParamName}}"]) {
+            headers['{{keyParamName}}'] = this.APIConfiguration.apiKeys["{{keyParamName}}"];
+        }
+{{/isKeyInHeader}}
+{{#isKeyInQuery}}
+        if (this.APIConfiguration.apiKeys["{{keyParamName}}"]) {
+            queryParameters.set('{{keyParamName}}', this.APIConfiguration.apiKeys["{{keyParamName}}"]);
+        }
+
+{{/isKeyInQuery}}
+{{/isApiKey}}
+{{#isBasic}}
+        if (this.APIConfiguration.username || this.APIConfiguration.password) {
+            headers['Authorization'] = btoa(this.APIConfiguration.username + ':' + this.APIConfiguration.password);
+        }
+
+{{/isBasic}}
+{{#isOAuth}}
+        if (this.APIConfiguration.accessToken) {
+            let accessToken = typeof this.APIConfiguration.accessToken === 'function'
+                ? this.APIConfiguration.accessToken()
+                : this.APIConfiguration.accessToken;
+            headers['Authorization'] = 'Bearer ' + accessToken;
+        }
+{{/isOAuth}}
+{{/authMethods}}
+        {{^produces}}
+        headers['Accept'] = 'application/json';
+        {{/produces}}
+        {{#produces.0}}
+        headers['Accept'] = '{{{mediaType}}}';
+        {{/produces.0}}
+{{#bodyParam}}
+        {{^consumes}}
+        headers['Content-Type'] = 'application/json';
+        {{/consumes}}
+        {{#consumes.0}}
+        headers['Content-Type'] = '{{{mediaType}}}';
+        {{/consumes.0}}
+{{/bodyParam}}
+
+{{#hasFormParams}}
+        let formData: FormData = new FormData();
+        headers['Content-Type'] = 'application/x-www-form-urlencoded;charset=UTF-8';
+{{#formParams}}
+        {{#isListContainer}}
+        if ({{paramName}}) {
+        {{#isCollectionFormatMulti}}
+            {{paramName}}.forEach((element) => {
+                formData.append('{{baseName}}', <any>element);
+            })
+        {{/isCollectionFormatMulti}}
+        {{^isCollectionFormatMulti}}
+            formData.append('{{baseName}}', {{paramName}}.join(COLLECTION_FORMATS['{{collectionFormat}}']));
+        {{/isCollectionFormatMulti}}
+        }
+        {{/isListContainer}}
+        {{^isListContainer}}
+        if ({{paramName}} !== undefined) {
+            formData.append('{{baseName}}', <any>{{paramName}});
+        }
+        {{/isListContainer}}
+{{/formParams}}
+
+{{/hasFormParams}}
+
+        return this.httpClient.{{httpMethod}}(`${this.basePath}{{{path}}}{{#hasQueryParams}}?${encodeURIComponent(queryParameters.join('&'))}{{/hasQueryParams}}`{{#bodyParam}}, {{paramName}} {{/bodyParam}}{{#hasFormParams}}, body{{/hasFormParams}}, headers)
```

Comment:
`encodeURIComponent(queryParameters.join('&'))`` should be `queryParameters.join('&')`, otherwise the `=` between the parameter name and the value will be converted to `%3D`